yarn/src/main/scala/org/apache/spark/deploy/yarn/ClientArguments.scala

```diff
@@ -87,6 +93,24 @@ private[spark] class ClientArguments(args: Array[String], sparkConf: SparkConf)
       throw new IllegalArgumentException(
         "You must specify at least 1 executor!\n" + getUsageMessage())
     }
+    if (isClusterMode) {
+      for (key <- Seq(amMemKey, amMemOverheadKey)) {
+        if (sparkConf.getOption(key).isDefined) {
+          println(s"$key is set but does not apply in cluster mode.")
+        }
+      }
+    } else {
+      for (key <- Seq(driverMemKey, driverMemOverheadKey)) {
+        if (sparkConf.getOption(key).isDefined) {
+          println(s"$key is set but does not apply in client mode.")
+        }
+      }
+      if (isDriverMemSet) {
+        println("--driver-memory is set but does not apply in client mode.")
+      }
```

Comment:
This warning doesn't make sense. It's a perfectly reasonable thing for YARN users to set the driver memory in client mode.